pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotRealtimeTableResource.java

```diff
@@ -78,13 +79,21 @@ public Response pauseConsumption(
   @Path("/tables/{tableName}/resumeConsumption")
   @Produces(MediaType.APPLICATION_JSON)
   @ApiOperation(value = "Resume consumption of a realtime table",
-      notes = "Resume the consumption for a realtime table")
+      notes = "Resume the consumption for a realtime table. If offset criteria is not provided, consumption continues "
+          + "based on the offsets in segment ZK metadata")
   public Response resumeConsumption(
-      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName) {
+      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
+      @ApiParam(value = "smallest | largest") @QueryParam("offsetCriteria") String offsetCriteria) {
```

Comment:
Instead of `offsetCriteria`, can we call it `consumeFrom`? We can translate it to whatever we want internally.

And the possible values can be `forceEarliest`, `forceLatest`, or `best` (yes, we need a third value).

`forceEarliest` and `forceLatest` will ignore any previous completed segment offsets and just do as  forced. Pick the earliest or latest offsets and resume consumption.

The `best` option will attempt to minimize data loss, pick up the first available event _after_ the last consumed event.

Reply:
👍 on `consumeFrom`. It reads better that way, but I prefer to keep `smallest` and `largest` so the uri looks like:
`/tables/<tableName>/resumeConsumption?consumeFrom=smallest`

For the behavior of the `best` option you suggested, we currently achieve that if `consumeFrom` parameter is not specified.

Reply:
Let us add all three options, and say that the default is `best`. Also, I prefer earliest and latest as being more intuitive. Alternative: `oldestEvents` `mostRecentEvents`, `fromLastPause`

Reply:
Done.